drools-wb-services/drools-wb-verifier/drools-wb-verifier-api/src/main/java/org/drools/workbench/services/verifier/api/client/relations/ConflictResolver.java

```diff
@@ -22,87 +22,89 @@
 public class ConflictResolver {
 
     private final InspectorList collection;
-    private final Conflicts     conflicts;
+    private final Conflicts conflicts;
 
-    public ConflictResolver( final InspectorList list,
-                             final boolean record ) {
+    public ConflictResolver(final InspectorList list,
+                            final boolean record) {
         this.collection = list;
-        conflicts = new Conflicts( record );
+        conflicts = new Conflicts(record);
     }
 
-    private Conflict isConflicting( final InspectorList otherCollection ) {
+    private Conflict isConflicting(final InspectorList otherCollection) {
 
-        if ( collection == null || otherCollection == null ) {
+        if (collection == null || otherCollection == null) {
             return Conflict.EMPTY;
         }
 
-        for ( Object o : collection ) {
-            if ( o instanceof IsConflicting ) {
-                final Conflict conflict = hasConflictingObjectInList( otherCollection,
-                                                                      ( IsConflicting ) o );
-                if ( conflict.foundIssue() ) {
-                    return new Conflict( collection,
-                                         otherCollection,
-                                         conflict );
+        for (Object o : collection) {
+            if (o instanceof IsConflicting) {
+                final Conflict conflict = hasConflictingObjectInList(otherCollection,
+                                                                     (IsConflicting) o);
+                if (conflict.foundIssue()) {
+                    return new Conflict(collection,
+                                        otherCollection,
+                                        conflict);
                 }
             }
         }
 
         return Conflict.EMPTY;
     }
 
-    private static Conflict getConflictingObjects( final InspectorList collection,
-                                                   final IsConflicting isConflicting ) {
+    private static Conflict getConflictingObjects(final InspectorList collection,
+                                                  final IsConflicting isConflicting) {
 
-        if ( isConflicting == null || collection == null ) {
+        if (isConflicting == null || collection == null) {
             return Conflict.EMPTY;
         }
 
-        for ( final Object other : collection ) {
-            return isConflicting( isConflicting,
-                                  ( HasUUID ) other );
+        for (final Object other : collection) {
+            final Conflict conflicting = isConflicting(isConflicting,
+                                                       (HasUUID) other);
+            if (conflicting.foundIssue()) {
+                return conflicting;
+            }
```

Comment:
The fix is here. We need to check all the items for an conflict.

Reply:
I just wonder if the first `conflicting` that has `foundIssue() == true` is enough to return. As the method `getConflictingObjects` suggests me some list of conflicting objects should be returned? Shouldn't be the list `InspectorList collection` iterated always and collect all `conflicting` that has `foundIssue() == true`? Is my question a little bit clear?

Reply:
The `conflicting` contains the two conflicting items, what ever they might be. To save time I just pick up the first issue.

Some additional information:  Then on reruns I check if this is still true. If it is then we skip the iteration all together and save some time. Conflicts are not always a bad thing. In this case we were checking for conflicts during subsumption checks. If there is a conflict the two items can not subsume each other. Subsumption is the heaviest operation, but it should also be the most rare situation in the table. Most tables should contain conflicts in the LHS, but this is different to conlicting rules. Where conflicting rules means two rules harm each other, conflicting LHS means the rules just fire with different Patterns.

You might have known all that. Just throwing as much info about this as possible :)

Reply:
Thanks Toni, I see you understand the best your code, it is a little bit more clear to me after your explanation.